scripts/mark-definite-framework-results.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+"""
+Uses a number of supplied json schemas to determine whether suppliers on a particular framework should be marked as
+onFramework. Attemots to recover a less strict subset of the schema from inside the declaration_definite_pass_schema
+at json path definitions/notDefiniteFail. This is used to differentiate between suppliers that definitely fail and those
+that will require a human decision ("discretionary"). If not found, all suppliers failing
+declaration_definite_pass_schema will have onFramework left unmodified (probably remaining null).
+
+Will also mark draft services with their respectivee validity (in the status field) as determined by
+draft_service_schema.
+
+Usage: mark-definite-framework-results.py [options] <stage> <api_token> <framework_slug>
+            <declaration_definite_pass_schema_path> [<draft_service_schema_path>]
+
+--updated-by=<user_string>        Specify updated_by string to use in API requests
+--dry-run                         Don't actually perform any updates on API
+--reassess-passed-sf              Don't skip supplier_frameworks with onFramework already True
+--reassess-failed-sf              Don't skip supplier_frameworks with onFramework already False
+--reassess-failed-draft-services  Don't skip draft_services with "failed" status
+-v, --verbose                     Produce more detailed console output
```

Comment:
Running locally with `-v` I get the following error for one of the suppliers, which prevents it from determining a result (runs fine without the `-v`):
```
2016-12-12 16:02:35,465 dmutils.logging ERROR failed to format log message: "u'status'" not found
Traceback (most recent call last):
  File "/Users/kevinkeenoy/gdsworkspace/digitalmarketplace-scripts/venv/lib/python2.7/site-packages/dmutils/logging.py", line 117, in format
    msg = msg.format(**record.__dict__)
KeyError: "u'status'"
2016-12-12 16:02:35,465 script DEBUG 		Failed service_schema @ : {u'status': u'submitted', u'frameworkFramework': u'dos', u'bespokeSystemInformation': True, u'supplierId': 700008, u'links': {u'self': u'http://localhost:5000/draft-services/194', u'copy': u'http://localhost:5000/draft-services/194/copy', u'complete': u'http://localhost:5000/draft-services/194/complete', u'publish': u'http://localhost:5000/draft-services/194/publish'}, u'lotSlug': u'digital-outcomes', u'helpGovernmentImproveServices': False, u'openStandardsPrinciples': True, u'updatedAt': u'2016-12-12T15:47:55.325566Z', u'frameworkSlug': u'digital-outcomes-and-specialists-2', u'locations': [u'Offsite', u'Scotland'], u'supplierName': u"Kev's Pies", u'frameworkStatus': u'open', u'dataProtocols': True, u'lotName': u'Digital outcomes', u'frameworkName': u'Digital Outcomes and Specialists 2', u'lot': u'digital-outcomes', u'performanceAnalysisTypes': [u'A/B and multivariate testing', u'Data analysis', u'Data cleansing', u'Data visualisation'], u'id': 194, u'createdAt': u'2016-12-12T15:47:35.110985Z'} is not valid under any of the given schemas
```
Unfortunately no more stack trace than that though.